lib/services/elasticsearch.js

```diff
@@ -1109,6 +1137,35 @@ class ElasticSearch extends Service {
     return this._mexecute(esRequest, extracted.documents, extracted.rejected);
   }
 
+  /**
+   * Create or replace multiple documents at once.
+   * Does not adds additional metadata
+   *
+   * @param  {Request} request - Kuzzle API request
+   * @return {Promise}
+   */
+  mwriteDocuments(request) {
+    const
+      esRequest = initESRequest(request, ['consistency', 'refresh', 'timeout', 'fields']),
+      extracted = extractMDocuments(request, {});
+
+    esRequest.body = [];
+
+    let i; // NOSONAR
+    for (i = 0; i < extracted.documents.length; i++) {
+      esRequest.body.push({
+        index: {
+          _index: request.input.resource.index,
+          _type: request.input.resource.collection,
+          _id: extracted.documents[i]._id
+        }
+      });
+      esRequest.body.push(extracted.documents[i]._source);
```

Comment:
Why not call `push` once ?

Reply:
Elasticsearch bulk syntax need the element N to be some kind of header and the element N+1 is actually the document body. 
https://doc.kuzzle.io/api/1/controller-bulk/import/

Reply:
Actually, it's faster to call 2 time `Array.push` with one element than 1 time with 2 elements:

<details>
<pre>
.add('2 push', () => {
  const tab = []
  for (let i = 0; i < 10000; ++i) {
    tab.push(elem);
    tab.push(elem2);
  }
})
.add('1 push', () => {
  const tab = [];
  for (let i = 0; i < 10000; ++i) {
    tab.push(elem, elem2)
  }
})
</pre>
<pre>
1 push x 5,475 ops/sec ±2.01% (91 runs sampled)
2 push x 10,486 ops/sec ±0.49% (96 runs sampled)
Fastest is 2 push
</pre>
</details>